### PR TITLE
FIRApp optimize config time

### LINKED ISSUE #6902: +[FIRApp registerInternalLibrary:withName:] take too much time
### Feature proposal

*   FIRApp: remove +load or reduce the time cost of [FIRApp registerInternalLibrary:withName:] 

As image shown below, [FIRApp registerInternalLibrary:withName:]  in App launch Initializing phase take 78ms, it is too much for our App launch! So, could you optimize this. 

Wish you accept this suggestion!
Screen Shot 2020-11-05 at 13 43 25

### PR DESCRIPTION
**Pull request by maksymmalyhin**
Fixes #6902.

- defer initialization of Firebase user agent
- move CoreDiagnostics call to a background thread

### CODE REVIEW

**Review by paulb777 — COMMENTED** [review]
LGTM, would like @ryanwilson's approval too for 7.1.0 since we're so late in the release cycle.

File: FirebaseCore/CHANGELOG.md
Comment (morganchen12) [inline FirebaseCore/CHANGELOG.md:2]: nit: Prefer Swift in release notes. Also updated wording a bit

```swift
- [fixed] Reduced `FirebaseApp.registerInternalLibrary(name:)` impact on app launch time.
```

**Review by morganchen12 — COMMENTED** [review]
Changelog LGTM with one comment, please wait for iCore approvers before merging.

File: FirebaseCore/CHANGELOG.md
Comment (maksymmalyhin) [reply FirebaseCore/CHANGELOG.md]: `+[FIRApp registerInternalLibrary:withName:]` is an internal method that is not available from Swift so using Swift version would be rather confusing. I applied the wording change and added the affected public API mention. PTAL

File: FirebaseCore/CHANGELOG.md
Comment (paulb777) [inline FirebaseCore/CHANGELOG.md:1]: 7.2.0

**Review by paulb777 — APPROVED** [review]

**Pull request merged**
